src/signature.rs

```diff
@@ -8,19 +8,36 @@
 // - Henry de Valence <hdevalence@hdevalence.ca>
 
 //! RedDSA Signatures
-use core::marker::PhantomData;
+use core::{fmt, marker::PhantomData};
 
 use crate::SigType;
 
 /// A RedDSA signature.
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[derive(Copy, Clone, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct Signature<T: SigType> {
     pub(crate) r_bytes: [u8; 32],
     pub(crate) s_bytes: [u8; 32],
     pub(crate) _marker: PhantomData<T>,
 }
 
+impl<T: SigType> fmt::Debug for Signature<T> {
+    #[cfg(feature = "alloc")]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Signature")
+            .field("r_bytes", &hex::encode(&self.r_bytes))
```

Comment:
Consider writing a helper, something like this:
```rust
#[cfg(feature = "alloc")]
fn hex_if_possible(bytes: &[u8]) -> String {
    &hex::encode(bytes)
}
#[cfg(not(feature = "alloc"))]
fn hex_if_possible(bytes: &[u8]) -> &[u8] {
    bytes
}
```

Reply:
Yeah and then invoke `hex_if_possible()` inside `.field()`, whichever one is available via the `alloc` feature

Reply:
Good idea, thanks! I considered something like this but didn't think of using functions with different return types. Done in 5a41fdb